### PR TITLE
Do not request data for invalid germline sample lists

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/3471.

Changes proposed in this pull request:
- Fixed by using `getAllSampleListsInStudyUsingGET` and filtering sample ids by `germline`.

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -608,17 +608,23 @@ export class ResultsViewPageStore {
...
-                return sampleIds.map(sampleId => ({sampleId, studyId}));
-            }));
+            if (this.germlineSampleLists.result) {
+                const ids: string[][] = await Promise.all(this.germlineSampleLists.result.map(sampleList => {
+                    return client.getAllSampleIdsInSampleListUsingGET({
+                        sampleListId: sampleList.sampleListId
+                    });
+                }));
+
+                return _.flatten(ids.map((sampleIds: string[], index: number) => {
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: this doesnt seem multiple-study-safe - need to perform conversion from sampleId to uniqueSampleKey, or just return Sample objects

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 2]
```diff
@@ -675,9 +681,35 @@ export class ResultsViewPageStore {
...
     }, {});
 
-    private getGermlineSampleListId(studyId:string):string {
-        return `${studyId}_germline`;
-    }
+    readonly sampleListIdsInStudies = remoteData({
+        await: () => [this.studies],
+        invoke: async () => {
+            const studies: string[] = this.studyIds;
+            return await Promise.all(studies.map(studyId => {
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: return _.flatten(await... ) would be more intuitive I think, to return a flat list

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 1 above)
Comment (onursumer) [reply src/pages/resultsView/ResultsViewPageStore.ts]: Does this mean that the current code is already not multiple-study-safe?

```JavaScript
return _.flatten(ids.map((sampleIds: string[], index: number) => {
    const studyId = studies[index];
    return sampleIds.map(sampleId => ({sampleId, studyId}));
}));
```

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts]: nevermind, I didn't read carefully enough. looks good

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 2 above)
Comment (onursumer) [reply src/pages/resultsView/ResultsViewPageStore.ts]: I did it this way to keep the current logic in `germlineConsentedSamples` intact. Seems like we are already doing the flattening there.

Comment (onursumer) [reply src/pages/resultsView/ResultsViewPageStore.ts]: Hmm, actually it looks okay to be flattened here...

**Comment by jjgao** [comment]
I think we may want to have a better solution. Let's discuss.

**Pull request closed without merging**